### PR TITLE
Mandelbrot: Various improvements

### PR DESCRIPTION
**Pull request by gunnarbeutner**
**Mandelbrot: Maintain aspect ratio when selecting a region**

This makes sure the aspect ratio of the widget and the selection match. Otherwise you'd end up with distorted images when zooming in.

**Mandelbrot: Implement color smoothing with gradients**

This removes the color banding that happens for some of the "outer" areas which all have the same iteration count.

Screenshot from 2021-05-17 17-07-01

**Mandelbrot: Add support for exporting the current image**

Unfortunately this means `unveil()` won't work - at least until we get something like `FilePickerServer`.

**Mandelbrot: Keep the aspect ratio when (re-)sizing the window**

Previously the initial aspect ratio was incorrect and there was nothing to ensure that the aspect ratio is kept when resizing the
window.

**Mandelbrot: Export images in a fixed resolution**
    
This makes the exported image independent from the current window size and just always exports it at 1920x1080.

### CODE REVIEW

File: Userland/Demos/Mandelbrot/Mandelbrot.cpp [hunk 1]
```diff
@@ -203,15 +209,25 @@ void Mandelbrot::resize_event(GUI::ResizeEvent& event)
...
 {
     auto app = GUI::Application::construct(argc, argv);
 
-    if (pledge("stdio recvfd sendfd rpath", nullptr) < 0) {
+    if (pledge("stdio thread recvfd sendfd rpath wpath cpath", nullptr) < 0) {
         perror("pledge");
         return 1;
     }
 
+#if 0
```
Comment (bgianfo) [inline Userland/Demos/Mandelbrot/Mandelbrot.cpp:230]: How about we just unveil the home directory for creation, and pop an error if they try to save somewhere else?

File: Userland/Demos/Mandelbrot/Mandelbrot.cpp [hunk 2]
```diff
@@ -203,15 +209,25 @@ void Mandelbrot::resize_event(GUI::ResizeEvent& event)
     m_set.resize(event.size());
 }
 
+void Mandelbrot::export_image(String const& export_path)
+{
+    auto png = Gfx::PNGWriter::encode(m_set.bitmap());
+    auto file = fopen(export_path.characters(), "wb");
```
Comment (bgianfo) [inline Userland/Demos/Mandelbrot/Mandelbrot.cpp:215]: If the open fails we silently claim successful export.

**Review by bgianfo — APPROVED** [review]

File: Userland/Demos/Mandelbrot/Mandelbrot.cpp (hunk 2 above)
Comment (gunnarbeutner) [reply Userland/Demos/Mandelbrot/Mandelbrot.cpp:215]: Fixed. :)

File: Userland/Demos/Mandelbrot/Mandelbrot.cpp (hunk 1 above)
Comment (gunnarbeutner) [reply Userland/Demos/Mandelbrot/Mandelbrot.cpp:234]: Unveiling `/home`, `/etc/group` and `/etc/passwd` sort of works. However, when the user navigates outside of `/home` they end up with an empty file list. Also, this assumed the user's home directory is in `/home`. I think I'd rather wait for `FilePickerServer`. :)

Comment (bgianfo) [reply Userland/Demos/Mandelbrot/Mandelbrot.cpp:234]: Sure, makes sense.

**Comment by gunnarbeutner** [comment]
I've added another commit that makes sure the aspect ratio is reasonable when (re-)sizing the window.

**Pull request merged**